Update docs around getting started in VS Code (#1034)
tern-tools/tern: CONTRIBUTING.md

### Visual Studio Code Development Container

Developing using Visual Studio Code development container lets you use a Docker container as a development environment. The Tern repository contains a definition for a pre-configured Docker container to simplify the setup process. To use it you need to download the following software on your host machine:

- [Visual Studio Code](https://code.visualstudio.com/)
- [Docker](https://www.docker.com/get-started)
- [The Visual Studio Code Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)

Once you have all that installed you can setup your dev enviornment:

1. Fork the main Tern repository using your GitHub account. Once forked, clone *your* fork (not the main Tern repo) on your local desktop.
2. Using the terminal, `cd` into the forked Tern directory on your local desktop and setup your git credentials:
```
    git config --global user.name "Your Name"
    git config --global user.email "your.email@address.com"
```

> 3. Open Visual Studio Code. Click the `><` icon in the lower lefthand corner. A search bar will come up at the top of the window. Type "Remote-Container: Open Folder in Container" and click on it. In the window that pops up, navigate to the cloned `tern` folder on you desktop and click "Open".

Reviewer: This the remote settings icon, but it's not where you find that command. They need to click on the gear/cog icon. 
```suggestion
3. Open Visual Studio Code. Click the `gear` icon in the lower lefthand corner, or use the `F1` key on your keyboard. This opens the "Command Palette". A search bar will come up at the top of the window. Type "Remote-Container: Open Folder in Container" and click on it. In the window that pops up, navigate to the cloned `tern` folder on you desktop and click "Open".
```

> 4. On the left-most side panel, click on the icon that looks like four squares with one detaching. A search bar will appear. Use this to install the Docker Extension. It might be suggested, but if not, you can type "Docker" to search for it.

Reviewer: ```suggestion
4. On the left-most side panel, click on the icon that looks like four squares with one detaching. This is the "Extensions" menu. A search bar will appear. Use this to install the Docker Extension. It might be suggested, but if not, you can type "Docker" to search for it. Make sure to reload once it's installed.
```

> 5. If you are working on a Mac, there is a reported [bug](https://github.com/microsoft/vscode-remote-release/issues/3149) and [workaround](https://github.com/microsoft/vscode-remote-release/issues/3149#issuecomment-646588532) when using the Docker extension with a remote container. To change the Docker path once the Docker extension is installed, click on the `><` icon, type "Settings" and update the `Docker Path` section to be `/usr/local/bin/docker`.

Reviewer: ```suggestion
5. If you are working on a Mac, there is a reported [bug](https://github.com/microsoft/vscode-remote-release/issues/3149) and [workaround](https://github.com/microsoft/vscode-remote-release/issues/3149#issuecomment-646588532) when using the Docker extension with a remote container. To change the Docker path once the Docker extension is installed, click on the `gear` icon in the lower lefthand corner, select "Settings", and update the `Docker: Docker Path` section to be `/usr/local/bin/docker`.
```

10. Create a new branch by running `git checkout -b <newbranch-name>`. You're now ready to start making changes to Tern, which you can do via the Terminal or using GUI editor provided by VS Code. Note that you will need to commit your changes with git using the Terminal command line.

### Setting up a Long Term Development Environment

You may have already cloned the project and started working on it. If you're reading this after the fact, I would highly recommend you save your work and set up a new development environment in this way.

1. Set up a Python virtual environment that has either the Python 3.6 or Python 3.7 executable. See [here](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments) for instructions on how to set this up using your host machine's Python3. There are also a guide to [managing virtual environments using pipenv](https://docs.python-guide.org/dev/virtualenvs/) but I haven't used it. Once done, you should have a folder created. Change to that folder.
2. Clone *your fork* of Tern in the virtual environment folder.

> 3. Activate your virtual environment `source bin/activate`. NOTE: This specific activate script only works for Bash shells. If you need to activate a Fish Shell or C Shell you should use `source/bin/activate.fish` or `source/bin/activate.csh`, respectively.

Reviewer: Could we update this section to remove "C Shell" and replace it with ZSH? ZSH is the default terminal on MacOS now, so this may be required if using the default doesn't automatically work.

Author: Folks who are using zsh can follow the regular directions to activate the virtualenv I believe so maybe my wording around 'bash shells" needs to change. If folks are using C Shell for whatever reason they will need to use the .csh activation script. I think it's still useful to keep this in here as an FYI?

Reviewer: Agreed
